src/gsy_e/models/strategy/energy_parameters/heatpump/pcm_tank/pcm_tank.py

```diff
@@ -0,0 +1,504 @@
+from collections import defaultdict
+import logging
+from dataclasses import dataclass
+from statistics import mean
+from typing import Dict, Union, Optional
+
+from gsy_framework.constants_limits import ConstSettings, GlobalConfig, FLOATING_POINT_TOLERANCE
+from gsy_framework.read_user_profile import InputProfileTypes
+from gsy_framework.utils import (
+    convert_kWh_to_W,
+    convert_W_to_kWh,
+    convert_kJ_to_kWh,
+    convert_pendulum_to_str_in_dict,
+    convert_str_to_pendulum_in_dict,
+)
+from pendulum import DateTime
+
+from gsy_e.models.strategy.energy_parameters.heatpump.constants import SPECIFIC_HEAT_CAPACITY_WATER
+from gsy_e.models.strategy.energy_parameters.heatpump.cop_models import (
+    COPModelType,
+    cop_model_factory,
+)
+from gsy_e.models.strategy.energy_parameters.heatpump.heat_pump import (
+    HeatPumpEnergyParameters,
+)
+from gsy_e.models.strategy.energy_parameters.heatpump.pcm_tank.pcm_models import (
+    PCMDischargeModel,
+    PCMChargeModel,
+)
+from gsy_e.models.strategy.energy_parameters.heatpump.pcm_tank.pcm_model_utils_constants import (
+    MASS_FLOW_RATE,
+    NUMBER_OF_PCM_ELEMENTS,
+)
+from gsy_e.models.strategy.state import HeatPumpState
+from gsy_e.models.strategy.state.heat_pump_state import delete_time_slots_in_state
+from gsy_e.models.strategy.state.base_states import StateInterface
+from gsy_e.models.strategy.strategy_profile import profile_factory
+from gsy_e import constants
+
+log = logging.getLogger()
+
+
+@dataclass
+class PCMTankParameters:
+    """Nameplate parameters of a water tank."""
+
+    min_temp_C: float = ConstSettings.HeatPumpSettings.MIN_TEMP_C
+    max_temp_C: float = ConstSettings.HeatPumpSettings.MAX_TEMP_C
+    initial_temp_C: float = ConstSettings.HeatPumpSettings.INIT_TEMP_C
+    max_capacity_kWh: float = 6.0  # todo: put default value somewhere
+
+
+class PCMTankState(StateInterface):
```

Comment:
This breaks a bit the file hierarchy, e.g. this class should be under `state` package. Having said that we can leave it here and refactor at a later stage.